specification/security/resource-manager/Microsoft.Security/preview/2024-02-01-preview/assessments.json

```diff
@@ -0,0 +1,578 @@
+        }
+      },
+      "required": [
+        "code"
+      ]
+    },
+    "AssessmentRisk": {
+      "type": "object",
+      "description": "The risk properties of the assessment",
+      "properties": {
+        "level": {
+          "type": "string",
+          "description": "The level of the risk",
+          "enum": [
+            "Low",
+            "Medium",
+            "High",
+            "Critical"
+          ],
+          "x-ms-enum": {
+            "name": "AssessmentRiskLevel",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Low",
+                "description": "Low risk"
+              },
+              {
+                "value": "Medium",
+                "description": "Medium risk"
+              },
+              {
+                "value": "High",
+                "description": "High risk"
+              },
+              {
+                "value": "Critical",
+                "description": "Critical risk"
+              }
+            ]
+          }
+        },
+        "attackPathsReferences": {
+          "type": "array",
+          "description": "List of Attack Path IDs that effect the risk for this assessment",
+          "items": {
+            "type": "string"
+          }
+        },
+        "riskFactors": {
+          "type": "array",
+          "description": "List of risk factors that were considered for calculating the risk",
+          "items": {
+            "type": "string"
+          }
+        },
+        "paths": {
+          "type": "array",
+          "description": "List of paths that were considered for calculating the risk",
+          "items": {
+            "$ref": "#/definitions/AssessmentRiskPath"
+          }
+        }
+      }
+    },
+    "AssessmentRiskPath": {
+      "type": "object",
+      "description": "A contextual path that was considered for calculating an assessment risk",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the path"
+        },
+        "nodes": {
+          "type": "array",
+          "description": "List of node IDs in the path",
+          "items": {
+            "$ref": "#/definitions/AssessmentRiskPathReferenceItem"
+          }
+        },
+        "edges": {
+          "type": "array",
+          "description": "List of edge IDs in the path",
+          "items": {
+            "$ref": "#/definitions/AssessmentRiskPathReferenceItem"
+          }
+        },
+        "insights": {
+          "type": "array",
+          "description": "List of insight IDs in the path",
+          "items": {
+            "$ref": "#/definitions/AssessmentRiskPathReferenceItem"
+          }
+        }
+      }
+    },
+    "AssessmentRiskPathReferenceItem": {
+      "type": "object",
+      "description": "A single item from a contextual path that was considered for calculating an assessment risk",
+      "properties": {
+        "id": {
```

Comment:
The id, type etc. are some of the standard ARM property names, which ARM as well as other RPs assume to be part of an ARM resource. So, unless it has to be this name of the property for some reason, please use some different name e.g., itemId or referenceItemId here.

Reply:
Hi, thank you for the comment 😃
The `risk` object (which is where the mentioned properties appear) will never be an independent ARM resource (given how the product and properties are designed). Furthermore, it will always be a property inside an ARM resource, which means that the "type" and "id" properties won't collide with the ARM mandatory properties.

The mentioned names are correlated to an external database that is external to customer, so for the customer benefit we would like to keep these names for easier correlation.